internal/pkg/cli/env_delete.go

```diff
@@ -348,6 +372,80 @@ func (o *deleteEnvOpts) ensureRolesAreRetained() error {
 	return nil
 }
 
+// emptyBuckets returns nil if buckets were deleted successfully. Otherwise, returns the error.
+func (o *deleteEnvOpts) emptyBuckets() error {
+	envStack, err := o.rg.GetResources(&resourcegroupstaggingapi.GetResourcesInput{
+		ResourceTypeFilters: []*string{aws.String("cloudformation")},
+		TagFilters: []*resourcegroupstaggingapi.TagFilter{
+			{
+				Key:    aws.String(deploy.EnvTagKey),
+				Values: []*string{aws.String(o.name)},
+			},
+			{
+				Key:    aws.String(deploy.AppTagKey),
+				Values: []*string{aws.String(o.appName)},
+			},
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("find env stack resource: %w", err)
+	}
+	if len(envStack.ResourceTagMappingList) > 1 {
+		return fmt.Errorf("ambiguous env stacks found")
```

Comment:
I don't remember how exactly `GetResources` behave. If we have a service stack in the environment, it'll get tagged with app, env and svc. Could `GetResources` have returned two stacks, one env stack, and one svc stack?

Reply:
Is there a scenario where there can be a svc stack while the env stack is getting deleted? If so I need to filter out stacks tagged with a service somehow.

Reply:
ah, good point! 

I don't recall when we error out if there is still a lingering svc stack, but we def. do so sooner or later during the process. If it happens earlier than the execution of this line, then it is totally fair for us to expect only one stack.

Reply:
```
	//    * If you specify more than one filter in a single request, the response
	//    returns only those resources that satisfy all filters.
	//
	//    * If you specify a filter that contains more than one value for a key,
	//    the response returns resources that match any of the specified values
	//    for that key.
```
This is actually how it'll work: basically it won't include any svc stack

Reply:
But overall I am not too worried about this. The only fb is we can improve this error msg by telling them what stacks we found (ideally this should never happen)

Reply:
I agree, theoretically this should never happen, (which is why I didn't write a test for it) but we should handle the error in case of weird bugs 🐛 or catastrophe. I can rewrite it to be more verbose, but I really don't think that's necessary.

Reply:
No longer using `GetResources` to retrieve the environment stack!